### PR TITLE
New Emergency Shutter Sprites

### PR DESCRIPTION
**Pull request by nauticall**
Emergency shutters that shut. Oh, and flash too. Adds newfangled emergency shutters to the game, complete with flashing lights (that look eerily pretty in the dark) and hot/cold warning lights that flash when, you guessed it, hostile temperatures are on the other side of the door. Once again can you stare in awe at the one door that's protecting you from a painful death.

The pressure warning (amber/orange) alternates flashing with the red/blue (hot/cold) warning lights so you can get both warnings known to you at the same time. Maybe it might be extremely hot _and_ at dangerously high pressure on the other side too.

image
image
image
image
image
image

### CODE REVIEW

**Review by NonQueueingMatt — APPROVED** [review]

**Comment by NonQueueingMatt** [comment]
!review

File: code/game/machinery/doors/firedoor.dm [hunk 1]
```diff
@@ -8,7 +8,7 @@
 
 /obj/machinery/door/firedoor
 	name = "emergency shutter"
-	desc = "An airtight emergency shutter. Capable of sealing off breached areas."
+	desc = "An airtight emergency bulkhead designed to seal off areas from hostile environments. It flashes a warning light if it detects an environmental hazard on the other side."
```
Comment (SleepyGemmy) [inline code/game/machinery/doors/firedoor.dm:11]:
```suggestion
	desc = "An airtight emergency shutter designed to seal off areas from hostile environments. It flashes a warning light if it detects an environmental hazard on any side."
```

File: code/game/machinery/doors/firedoor.dm [hunk 2]
```diff
@@ -154,7 +154,7 @@
 		return
 
 	if(pdiff >= FIREDOOR_MAX_PRESSURE_DIFF)
-		to_chat(user, "<span class='warning'>WARNING: Current pressure differential is [pdiff]kPa! Opening door may result in injury!</span>")
+		to_chat(user, "<span class='danger'>WARNING: Current pressure differential is [pdiff]kPa! Opening door may result in injury!</span>")
```
Comment (SleepyGemmy) [inline code/game/machinery/doors/firedoor.dm:157]:
```suggestion
		to_chat(user, SPAN_DANGER("Current pressure differential is [pdiff] kPa. Opening door will likely result in injury."))
```

File: html/changelogs/nauticall-newfiredoors.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+# Optional: Remove this file after generating master changelog.  Useful for PR changelogs that won't get used again.
+delete-after: True
+
+# Any changes you've made.  See valid prefix list above.
+# INDENT WITH TWO SPACES.  NOT TABS.  SPACES.
+# SCREW THIS UP AND IT WON'T WORK.
+# Also, all entries are changed into a single [] after a master changelog generation. Just remove the brackets when you add new entries.
+# Please surround your changes in  double quotes ("), as certain characters otherwise screws up compiling. The quotes will not show up in the changelog.
+changes:
+  - rscadd: "Added new emergency shutters, both 1x1 and 2x1 versions. They now flash, too, depending on the danger they're shielding you against, just like before: orange for pressure, red for extreme heat, blue for extreme cold."
```
Comment (SleepyGemmy) [inline html/changelogs/nauticall-newfiredoors.yml:41]:
```suggestion
  - imageadd: "Added new emergency shutter sprites, both 1x1 and 2x1 versions. They now flash again, depending on the danger they're shielding you against, just like before: orange for pressure, red for extreme heat, and blue for extreme cold."
```

**Review by SleepyGemmy — APPROVED** [review]

**Review by FluffyGhoster — APPROVED** [review]

**Pull request merged**